homeUi/src/main/kotlin/com/gravatar/app/homeUi/presentation/home/gravatar/components/AvatarsGridSection.kt

```diff
@@ -0,0 +1,125 @@
+package com.gravatar.app.homeUi.presentation.home.gravatar.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyGridScope
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.lazy.grid.rememberLazyGridState
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.gravatar.app.homeUi.R
+import com.gravatar.app.homeUi.presentation.home.gravatar.AvatarUi
+import com.gravatar.restapi.models.Avatar
+import com.gravatar.ui.GravatarTheme
+import java.net.URI
+
+internal fun LazyGridScope.avatarsGridSection(
+    avatars: List<AvatarUi>,
+) {
+    item(
+        span = { GridItemSpan((maxLineSpan)) },
+    ) {
+        Text(
+            text = stringResource(R.string.gravatar_tab_previous_avatars),
+            style = MaterialTheme.typography.titleMedium,
+        )
+    }
+    item(
+        span = { GridItemSpan(maxLineSpan) },
+    ) {
+        Text(
+            text = stringResource(R.string.gravatar_tab_tap_for_options),
+            style = MaterialTheme.typography.bodyMedium,
+            modifier = Modifier.padding(bottom = 10.dp)
+        )
+    }
+    if (avatars.isEmpty()) {
+        item(
+            span = { GridItemSpan(maxLineSpan) },
+        ) {
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(100.dp)
+            )
+        }
+    } else {
+        items(items = avatars, key = { it.avatarId }) { avatarModel ->
+            SelectableAvatar(
+                avatar = avatarModel,
+                size = avatarSize,
+                modifier = Modifier,
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(showBackground = true)
+private fun VerticalAvatarSectionPreview() {
+    GravatarTheme {
```

Comment:
This `GravatarTheme` is provided by the SDK and not the App's one. As the one from the app won't be accessible here, we should probably consider extracting it to a different module. (In a different task / PR).

In this case, I would keep just `MaterialTheme`:
```suggestion
    MaterialTheme {
```